blockdevice/blockdevice.go

```diff
@@ -323,3 +330,6 @@ const (
 	// claiming
 	Unclaimed string = "Unclaimed"
 )
+
+// Hierarchy is the block device hierarchy on the system
```

Comment:
n_: this is a BlockDeviceCache which helps stores the discovered block devices in memory. This will help reduce the calls made to etcd for processing the disk events. The hierarchy information is secondary and is embedded into the block devices itself.

Reply:
@kmova .
> this is a BlockDeviceCache which helps stores the discovered block devices in memory.

Yes. We use this map to keep track of blockdevices present on the host in memory. 

> This will help reduce the calls made to etcd for processing the disk events. The hierarchy information is secondary and is embedded into the block devices itself.

This will not help is reduce the calls to etcd. This cache / Hierarchy struct helps to get the details of dependent devices without performing a rescan on the system for all the disk attributes. 
eg: If at a certain point, we are processing `sda1` and we need to get the UUID of its parent device. We will get `/dev/sda` is the parent of `/dev/sda1`. This will be used to query the cache and get/generate the UUID of `/dev/sda`.

Will add the above info in the comment.